src/fsipy/automatedPostprocessing/postprocessing_h5py/spectrograms.py

```diff
@@ -0,0 +1,747 @@
+        fluid_domain_id (int or list): ID of the fluid domain
+        solid_domain_id (int or list): ID of the solid domain
+
+    Returns:
+        tuple: (Processed data type, DataFrame, Case name, Image folder, Hi-pass visualization folder).
+    """
+    folder_path = Path(folder)
+    case_name = folder_path.parent.name
+    visualization_path = folder_path / "Visualization"
+
+    logging.info(f"--- Processing folder path {folder_path}\n")
+
+    mesh_name_suffix = "" if save_deg == 1 else "_refined"
+    mesh_path = Path(mesh_path)
+    mesh_path = mesh_path.with_name(f"{mesh_path.stem}{mesh_name_suffix}{mesh_path.suffix}")
+    mesh_path_fluid = mesh_path.with_name(f"{mesh_path.stem}_fluid.h5")  # Needed for formatting SPI data
+
+    formatted_data_folder_name = f"res_{case_name}_stride_{stride}t{start_t}_to_{end_t}save_deg_{save_deg}"
+    formatted_data_folder = folder_path / formatted_data_folder_name
+    visualization_separate_domain_folder = folder_path / "Visualization_separate_domain"
+    visualization_hi_pass_folder = folder_path / "Visualization_hi_pass"
+
+    image_folder = folder_path / "Spectrograms"
+    image_folder.mkdir(parents=True, exist_ok=True)
+
+    output_file_name = f"{case_name}_{quantity}_{component}.npz"
+    formatted_data_path = formatted_data_folder / output_file_name
+
+    logging.info("--- Preparing data")
+
+    # If the output file exists, don't re-make it
+    if formatted_data_path.exists():
+        logging.info(f'--- Formatted data already exists at: {formatted_data_path}\n')
+    else:
+        if quantity == "wss":
+            create_transformed_matrix(visualization_separate_domain_folder, formatted_data_folder, mesh_path_fluid,
+                                      case_name, start_t, end_t, quantity, fluid_domain_id, solid_domain_id, stride)
+        else:
+            # Make the output h5 files with quantity magnitudes
+            create_transformed_matrix(visualization_path, formatted_data_folder, mesh_path,
+                                      case_name, start_t, end_t, quantity, fluid_domain_id, solid_domain_id, stride)
+
+    logging.info("--- Reading data")
+
+    # For spectrograms, we only want the magnitude
+    df = read_npz_files(formatted_data_path)
+
+    logging.info("\n--- Processing data and getting ID's")
+
+    # We want to find the points in the sac, so we use a sphere to roughly define the sac.
+    x_sphere, y_sphere, z_sphere, r_sphere = fsi_region
+    sac_center = np.array([x_sphere, y_sphere, z_sphere])
+
+    if quantity == "wss":
+        wss_output_file = visualization_separate_domain_folder / "WSS_ts.h5"
+        surface_elements, coords = get_surface_topology_coords(wss_output_file)
+    else:
+        coords = get_coords(mesh_path)
+
+    if sampling_region == "sphere":
+        # Get wall and fluid ID's
+        fluid_ids, wall_ids, all_ids = get_domain_ids(mesh_path, fluid_domain_id, solid_domain_id)
```

Comment:
`wall_ids` should be renamed as `solid_ids` to make is consistent to the naming of `solid_domain_id`
Subsequently, all the other `wall_ids` need to be renamed as well.,
```suggestion
        fluid_ids, solid_ids, all_ids = get_domain_ids(mesh_path, fluid_domain_id, solid_domain_id)
```